x-pack/plugins/watcher/public/sections/watch_status/components/watch_history.tsx

```diff
@@ -159,112 +179,154 @@ const WatchHistoryUi = ({ intl, watchId }: { intl: InjectedIntl; watchId: string
   };
 
   const toggleWatchActivation = async () => {
-    try {
-      if (isActivated) {
-        await deactivateWatch(watchId);
-      } else {
-        await activateWatch(watchId);
-      }
+    const toggleActivation = isActivated ? deactivateWatch : activateWatch;
 
-      setIsActivated(!isActivated);
-    } catch (e) {
-      if (e.data.statusCode !== 200) {
-        toastNotifications.addDanger(
-          i18n.translate(
-            'xpack.watcher.sections.watchList.deactivateWatchErrorNotification.descriptionText',
-            {
-              defaultMessage: "Couldn't deactivate watch",
-            }
-          )
-        );
-      }
-    }
-  };
+    setIsTogglingActivation(true);
 
-  let flyout;
+    const { error } = await toggleActivation(loadedWatch.id);
 
-  if (detailWatchId !== undefined && watchHistoryDetails !== undefined) {
-    const detailColumns = [
-      {
-        field: 'id',
-        name: i18n.translate('xpack.watcher.sections.watchHistory.watchActionStatusTable.id', {
-          defaultMessage: 'Name',
-        }),
-        sortable: true,
-        truncateText: true,
-        render: (id: string) => {
-          return <EuiText>{id}</EuiText>;
-        },
-      },
-      {
-        field: 'state',
-        name: i18n.translate('xpack.watcher.sections.watchHistory.watchActionStatusTable.state', {
-          defaultMessage: 'State',
-        }),
-        sortable: true,
-        truncateText: true,
-        render: (state: string) => <WatchStatus status={state} />,
-      },
-    ];
+    setIsTogglingActivation(false);
 
-    flyout = (
-      <EuiFlyout
-        data-test-subj="indexDetailFlyout"
-        onClose={() => setDetailWatchId(undefined)}
-        aria-labelledby="indexDetailsFlyoutTitle"
-        maxWidth={600}
-      >
-        <EuiFlyoutHeader>
-          <EuiTitle size="s">
-            <h3>
-              <FormattedMessage
-                id="xpack.watcher.sections.watchHistory.watchHistoryDetail.title"
-                defaultMessage="Executed on {date}"
-                values={{ date: watchHistoryDetails.startTime }}
-              />
-            </h3>
-          </EuiTitle>
-        </EuiFlyoutHeader>
+    if (error) {
+      return toastNotifications.addDanger(
+        i18n.translate(
```

Comment:
I think we should refactor this so that we're switching between two complete messages: "Couldn't activate watch" and "Couldn't deactivate watch". This will give the translators full context and will also make it easier for them to structure each message differently in case one of the sentences doesn't follow the format `{auxiliary verb} {verb} {noun}` in the localized language.

Reply:
good call, fixed